nemo/collections/nlp/models/language_modeling/megatron_gpt_model.py

```diff
@@ -304,14 +310,19 @@ def training_step(self, batch, batch_idx):
         lr = self._optimizer.param_groups[0]['lr']
         self.log('lr', lr, rank_zero_only=True)
         self.log('global_step', self.trainer.global_step, prog_bar=True, rank_zero_only=True)
-        # TODO: make sure compute_consumed_samples works for pipeline parallelism
+        self.log('num_micro_batches', get_num_microbatches(), prog_bar=False, rank_zero_only=True)
+        consumed_samples = self.compute_consumed_samples(self.trainer.global_step + 1 - self.init_global_step)
```

Comment:
Was the +1 missing here a mistake earlier? Based on a fully converged mbart checkpoint, it looks like the number of consumed samples for the final number of steps (1M) checks out `megatron_bart--val_loss=0.78-step=1000000-consumed_samples=1024000000.0.ckpt`.

Reply:
`trainer.global_step` doesn't get updated until after `training_step`.  So the first time that this is called it will be `self.compute_consumed_samples(0)=0`. But it's being called after the first global batch has been processed.